src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -154,7 +154,8 @@ public static void BuildGraphNodeDependencies(List<AssociativeGraph.GraphNode> g
                             continue;
                         }
 
-                        currentNode.PushGraphNodeToExecute(gnode);
+                        currentNode.PushChildNode(gnode);
+                        gnode.PushParentNode(currentNode);
```

Comment:
We don't need two methods, setting parent could be done in `PushChildNode`, or setting child could be done in `PushParentNode`.